src/main/java/org/dependencytrack/event/kafka/processor/VulnerabilityScanResultProcessor.java

```diff
@@ -287,24 +345,300 @@ private Vulnerability syncVulnerability(final QueryManager qm, final Vulnerabili
      * @param component The {@link Component} to associate with
      * @param vulns     The {@link Vulnerability}s to associate with
      * @param scanner   The {@link Scanner} that identified the association
-     * @return A {@link Set} of {@link Vulnerability}s that were not previously associated with the {@link Component}
+     * @return A {@link List} of {@link UUID}s, for {@link Vulnerability}s that were not previously
+     * associated with the {@link Component}, and which have not been suppressed via {@link VulnerabilityPolicy}.
      */
-    private Set<Vulnerability> addVulnerabilities(final QueryManager qm, final Component component,
-                                                  final Set<Vulnerability> vulns, final Scanner scanner) {
-        final var newVulns = new HashSet<Vulnerability>();
-
-        for (final Vulnerability vuln : vulns) {
-            qm.runInTransaction(() -> {
-                if (!qm.contains(vuln, component)) {
-                    component.addVulnerability(vuln);
-                    qm.getPersistenceManager().makePersistent(new FindingAttribution(
-                            component, vuln, convert(scanner), null, null));
-                    newVulns.add(vuln);
+    private List<UUID> synchronizeFindingsAndAnalyses(final QueryManager qm, final Component component,
```

Comment:
might be good to refactor this one to smaller methods later

Reply:
Yep, more than fair point. I'll have another look at this today and see if I can untangle this mess.

Reply:
@mehab I separated the logic of *applying* the analyses into another method. That new method is still too large, but I need more time to figure out a better way to structure it properly... Please have another look, hope that I didn't make it worse at least! :D